src/main/java/org/jenkins/plugins/lockableresources/actions/LockableResourcesRootAction.java

```diff
@@ -137,15 +148,43 @@ public int getFreeResourcePercentage(String label) {
    * @return All possible labels.
    */
   public Set<String> getAllLabels() {
```

Comment:
check where it is used and set it to deprecated
This operation will be very slowly (because of synchronized over all resources)